File: android-designsystem/src/main/kotlin/com/thomaskioko/tvmaniac/compose/components/BadgeChip.kt
package com.thomaskioko.tvmaniac.compose.components

import androidx.compose.foundation.layout.padding
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.PreviewWrapper
import com.thomaskioko.tvmaniac.compose.components.ThemePreviews
import com.thomaskioko.tvmaniac.compose.components.TvManiacPreviewWrapperProvider
import com.thomaskioko.tvmaniac.compose.theme.TvManiacSpacing

@Composable
public fun PremiereBadge(
    modifier: Modifier = Modifier,
    text: String,
) {
    Surface(
        modifier = modifier,
        shape = MaterialTheme.shapes.small,
        color = MaterialTheme.colorScheme.onSurface,
    ) {
        Text(
            text = text,
            style = MaterialTheme.typography.labelSmall,
            color = MaterialTheme.colorScheme.background,
            modifier = Modifier.padding(horizontal = TvManiacSpacing.xSmall, vertical = TvManiacSpacing.xxSmall),
        )
    }
}

@Composable
public fun NewBadge(
    modifier: Modifier = Modifier,
    text: String,
) {
    Surface(
        modifier = modifier,
        shape = MaterialTheme.shapes.small,
        color = MaterialTheme.colorScheme.secondary,
    ) {
        Text(
            text = text,
            style = MaterialTheme.typography.labelSmall,
            color = MaterialTheme.colorScheme.onSecondary,
            modifier = Modifier.padding(horizontal = TvManiacSpacing.xSmall, vertical = TvManiacSpacing.xxSmall),
        )
    }
}

@ThemePreviews
@PreviewWrapper(TvManiacPreviewWrapperProvider::class)
@Composable
private fun PremiereBadgePreview() {
    PremiereBadge(text = "Premiere")
}

@ThemePreviews
@PreviewWrapper(TvManiacPreviewWrapperProvider::class)
@Composable
private fun NewBadgePreview() {
    NewBadge(text = "New")
}

File: ios/Modules/SwiftUIComponents/Sources/SwiftUIComponents/Components/BadgeChip.swift
import DesignSystem
import SwiftUI

public struct PremiereBadge: View {
    @Environment(\.appTheme) private var theme

    private let text: String

    public init(text: String) {
        self.text = text
    }

    public var body: some View {
        Text(text)
            .textStyle(theme.typography.labelSmall)
            .foregroundStyle(theme.colors.background)
            .padding(.horizontal, theme.spacing.xSmall)
            .padding(.vertical, theme.spacing.xxSmall)
            .background(
                RoundedRectangle(cornerRadius: theme.shapes.small)
                    .fill(theme.colors.onSurface)
            )
    }
}

public struct NewBadge: View {
    @Environment(\.appTheme) private var theme

    private let text: String

    public init(text: String) {
        self.text = text
    }

    public var body: some View {
        Text(text)
            .textStyle(theme.typography.labelSmall)
            .foregroundStyle(theme.colors.onSecondary)
            .padding(.horizontal, theme.spacing.xSmall)
            .padding(.vertical, theme.spacing.xxSmall)
            .background(
                RoundedRectangle(cornerRadius: theme.shapes.small)
                    .fill(theme.colors.secondary)
            )
    }
}

#Preview {
    VStack(spacing: 12) {
        PremiereBadge(text: "Premiere")
        NewBadge(text: "New")
    }
}
